src/sortStudents.ts

```diff
@@ -1,16 +1,59 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studArr = [...students];
+
+  function calculateAverage(grades: number[]): number {
+    return grades
+      .reduce((sum, val) => sum + val, 0) / grades.length;
+  }
+
+  function localeCompare(a: string, b: string): number {
+    return order === 'asc' ? a.localeCompare(b) : b.localeCompare(a);
+  }
+
+  function numberOrderHelper(a: number, b: number): number {
+    return order === 'asc' ? a - b : b - a;
+  }
 
+  studArr.sort((a: Student, b: Student) => {
+    switch (sortBy) {
+      default:
+        return 0;
+      case SortType.Name:
+        return localeCompare(a.name, b.name);
+      case SortType.Surname:
+        return localeCompare(a.surname, b.surname);
```

Comment:
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return localeCompare(a[sortBy], b[sortBy]);
```
You can combine the cases as follow